Assets/MRTK/SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs

```diff
@@ -425,8 +425,10 @@ void IMixedRealityTouchHandler.OnTouchCompleted(HandTrackingInputEventData event
         /// </summary>
         /// 
         public Vector3 GetWorldPositionAlongPushDirection(float localDistance)
-        {
-            float distance = (distanceSpaceMode == SpaceMode.Local) ? localDistance * LocalToWorldScale : localDistance;
+        {   
+            // We clamp the calculated local distance to the maxPushDistance,
+            // because LocalToWorldScale may return NaNs when object scale is zero.
+            float distance = Mathf.Clamp(0f, maxPushDistance, (distanceSpaceMode == SpaceMode.Local) ? localDistance * LocalToWorldScale : localDistance);
```

Comment:
Nit: For readability across it might be helpful to instead have something like this:

```c#
float distance = (distanceSpaceMode == SpaceMode.Local) ? localDistance * LocalToWorldScale : localDistance);
float clampedDistance = Mathf.Clamp(distance, 0f, maxPushDistance);
```

I think that the order of parameters is already kinda hard to remember (i.e. given @keveleigh's comment) so having a less operation-intensive thing on a single line could help to alleviate that)